### ISSUE TITLE
Unable to build app for iOS on macOS Catalina

### ISSUE THREAD

**Issue by sumanthratna** [flutter/flutter#58323]
## Steps to Reproduce

```zsh
flutter create --org com.o-wth mytest
cd mytest
flutter build ios
```

## Logs
[logs.txt](https://github.com/flutter/flutter/files/4706038/logs.txt)

`flutter doctor -v`:
```
[✓] Flutter (Channel stable, v1.17.2, on Mac OS X 10.15.5 19F96, locale en-US)
    • Flutter version 1.17.2 at /Users/suman/flutter
    • Framework revision 5f21edf8b6 (2 days ago), 2020-05-28 12:44:12 -0700
    • Engine revision b851c71829
    • Dart version 2.8.3

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/suman/android-sdks
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/suman/android-sdks
    • ANDROID_SDK_ROOT = /Users/suman/android-sdks
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[!] Connected device
    • Error: Sumanth Ratna is not connected. Xcode will continue when Sumanth Ratna is connected. (code -13)

! Doctor found issues in 1 category.
```

**Comment by VladyslavBondarenko** [comment]
Hi @sumanthratna 
I can see in logs 
```
If you are certain that Xcode supports development on this device, try disconnecting and reconnecting the device
```
And in doctor
```
Error: Sumanth Ratna is not connected. Xcode will continue when Sumanth Ratna is connected. (code -13)
```

Does https://stackoverflow.com/questions/26791477/xcode-device-locked-when-iphone-is-unlocked help?

**Comment by sumanthratna** [comment]
Thanks for your reply. Unfortunately, I'm unable to fix the issue. Originally, I was not building with a connected device (no Simulator or physical phone). 

I connected my phone (which is the device Sumanth Ratna), and then I got this error: `xcodebuild[39345:176559] [MT] iPhoneConnect: Failed _shouldMakeReadyForDevelopment check even though device is not locked by passcode.`. I tried https://stackoverflow.com/a/61636074/7127932, https://github.com/flutter/flutter/issues/41198#issuecomment-540131788, and the link you sent; none fixed the issue. 

`flutter build apk` works fine. 

Here's the output of `flutter build ios --verbose` with a Simulator running:  [verbose.txt](https://github.com/flutter/flutter/files/4711681/verbose.txt)

**Comment by sumanthratna** [comment]
I came across and https://github.com/flutter/flutter/issues/43799, and removing CPATH from my zshrc fixed this issue. Earlier, I had set CPATH so that I could build the Pillow Python library from source. 

I definitely think it would be convenient if flutter gracefully failed by detecting if CPATH is set, rather than bombarding me with tons of errors. :) 

I'm closing this since my original issue has been solved; let me know if I should reopen or create a new feature request ticket.
